app/models/beta/search/intercept_message.rb

```diff
@@ -10,14 +41,27 @@ class InterceptMessage
       def self.build(search_query)
         query = search_query.downcase
 
-        return unless query.eql?(I18n.t("#{query}.title"))
+        return if I18n.t("#{query}.message", default: nil).blank?
+
+        intercept_message = new
 
-        result = new
+        message = I18n.t("#{query}.message")
+        message = message.ends_with?('.') ? message : message.concat('.')
+
+        intercept_message.term = query
+        intercept_message.message = message
+
+        intercept_message
+      end
 
-        result.term = I18n.t("#{query}.title")
-        result.message = I18n.t("#{query}.message")
+      def formatted_message
+        return '' if message.blank?
 
-        result
+        GOODS_NOMENCLATURE_LINK_TRANSFORMERS.each_with_object(message.dup) do |(regex, transformer), transformed_message|
+          transformed_message.gsub!(regex) do |matched_text|
```

Comment:
maybe you don't need the bang version 
`.gsub!`
since you are using a block

Reply:
Using each with object so need to do this against the duplicate